backend/src/main/java/planit/people/preparation/APIs/API_Calendar.java

```diff
@@ -73,17 +78,70 @@ public ResponseEntity<List<DTO_PresetDetail>> getAllPresetFromPlanItUserId(@Path
     }
 
     /**
-     * NOT COMPLETE
+     * Update an already existing EventPreset
      *
      * @param dtoPresetDetail the body of the updated preset detail.
-     * @return Status OK
+     * @param planItUserId    the id of the PlanIt User who requested the update
+     * @return the updated preset detail
+     * @see Service_Calendar#upsertPresetDetail(Long, DTO_PresetDetail, Boolean)
      */
-    @PatchMapping
-            (path = "presets", consumes = {MediaType.APPLICATION_JSON_VALUE})
-    public ResponseEntity<Boolean> updateNewPreset(@RequestBody DTO_PresetDetail dtoPresetDetail) {
+    @PatchMapping(path = "presets/{planit-user-id}", consumes = {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<DTO_PresetDetail> updatePreset(@RequestBody DTO_PresetDetail dtoPresetDetail, @PathVariable("planit-user-id") Long planItUserId) {
         try {
             System.out.println("request: " + dtoPresetDetail);
-            serviceCalendar.upsertPresetDetail(null, dtoPresetDetail, true);
+            return new ResponseEntity<>(serviceCalendar.upsertPresetDetail(planItUserId, dtoPresetDetail, true), HttpStatus.OK);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    /**
+     * Delete an EventPreset instance for a given PlanItUser
+     *
+     * @param planItUserId the id of the PlanIt user whose instance should be removed.
+     * @param presetId     the id of the EventPreset which should be removed.
+     * @return HttpStatus.OK if deleted else HttpStatus.BAD_REQUEST
+     */
+    @DeleteMapping(path = "presets/{planit-user-id}/{preset-id}")
```

Comment:
Maybe let's give it some hierarchy?

Like `planit-user/{planit-user-id}/presets/{preset-id}`?

Reply:
The path parameters are defined in the documentation I do not see the point of adding extra strings

Reply:
The point of adding extra strings is to make it more understandable and to distinguish the hierarchical relationship of presets and users. It will also improve consistency as per the standards (https://restfulapi.net/resource-naming/):

```
2.1.3. store
A store is a client-managed resource repository. A store resource lets an API client put resources in, get them back out, and decide when to delete them.

A store never generates new URIs. Instead, each stored resource has a URI. The URI was chosen by a client when the resource initially put it into the store.

Use “plural” name to denote store resource archetype.

2.2.1. Use forward slash (/) to indicate hierarchical relationships
The forward-slash (/) character is used in the path portion of the URI to indicate a hierarchical relationship between resources. e.g.
```

btw this endpoint is not so complex, visiting the documentation should be the last resort